sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs

```diff
@@ -43,7 +43,12 @@ public KeyClient(Uri vaultUri, TokenCredential credential, KeyClientOptions opti
 
         public virtual Response<Key> CreateKey(string name, KeyType keyType, KeyCreateOptions keyOptions = default, CancellationToken cancellationToken = default)
         {
-            throw new NotImplementedException();
+            if (string.IsNullOrEmpty(name)) throw new ArgumentException($"{nameof(name)} can't be empty or null");
+            if (keyType == default) throw new ArgumentNullException(nameof(keyType));
+
+            var parameters = new KeyRequestParameters(keyType, keyOptions);
+
+            return SendRequest(HttpPipelineMethod.Put, parameters, () => new Key(name), cancellationToken, KeysPath, name);
```

Comment:
>return SendRequest(HttpPipelineMethod.Put, parameters, () => new Key(name), cancellationToken, KeysPath, name); [](start = 12, length = 111)

I missed this in the async review, but the path for create is actually /keys/{key-name}/create

Reply:
Are you sure? https://docs.microsoft.com/en-us/rest/api/keyvault/setsecret/setsecret

Reply:
Thank you! I fixed it with the versions url too